virtusize-auth/src/main/java/com/virtusize/android/auth/views/VitrusizeAuthActivity.kt

```diff
@@ -0,0 +1,254 @@
+package com.virtusize.android.auth.views
+
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.content.pm.ResolveInfo
+import android.net.Uri
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import androidx.browser.customtabs.CustomTabsIntent
+import androidx.browser.customtabs.CustomTabsService.ACTION_CUSTOM_TABS_CONNECTION
+import com.virtusize.android.auth.data.SnsType
+import com.virtusize.android.auth.data.VirtusizeUser
+import com.virtusize.android.auth.network.FacebookAPIService
+import com.virtusize.android.auth.network.GoogleAPIService
+import com.virtusize.android.auth.repositories.FacebookRepository
+import com.virtusize.android.auth.repositories.GoogleRepository
+import com.virtusize.android.auth.utils.VirtusizeAuthConstants.EXTRA_NAME_AUTH_URL
+import com.virtusize.android.auth.utils.VirtusizeAuthConstants.EXTRA_NAME_SNS_SCRIPT
+import com.virtusize.android.auth.utils.VirtusizeAuthConstants.NATIVE_APP_ID_KEY
+import com.virtusize.android.auth.utils.VirtusizeAuthConstants.SNS_ACCESS_TOKEN_KEY
+import com.virtusize.android.auth.utils.VirtusizeAuthConstants.SNS_CODE_KEY
+import com.virtusize.android.auth.utils.VirtusizeAuthConstants.SNS_ENV_KEY
+import com.virtusize.android.auth.utils.VirtusizeAuthConstants.SNS_REGION_KEY
+import com.virtusize.android.auth.utils.VirtusizeAuthConstants.SNS_TYPE_KEY
+import com.virtusize.android.auth.utils.VirtusizeUriHelper
+import com.virtusize.android.auth.utils.isVirtusizeSNSAuthURL
+import com.virtusize.android.util.valueOf
+import java.net.URLDecoder
+
+class VitrusizeAuthActivity : AppCompatActivity() {
```

Comment:
```suggestion
internal class VitrusizeAuthActivity : AppCompatActivity() {
```

Reply:
This class is kind of public, as it should be defined in the parent's app AndroidManifest. Even though the example works when this class is `internal` I think it's better to keep it public.

What do you think?

Reply:
If there's no need and we don't want the client to access VitrusizeAuthActivity in their Kotlin code, then keeping it internal would be better. If there's a chance that they will need to start the activity on their own, we can keep it public for now.

Reply:
Marked as `internal`